service/controller/resource/statuscondition/create.go

```diff
@@ -40,30 +40,42 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "found latest cluster")
 	}
 
+	var nodes []corev1.Node
+
 	tenantClient, err := r.tenantClient.K8sClient(ctx, cr)
-	if tenantclient.IsNotAvailable(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant client is not available yet")
+	if !tenantclient.IsNotAvailable(err) {
```

Comment:
I don't follow this change. Can you explain the reason to me? To me it looks like the code does the same but I would prefer the version we had. 

```
if tenantclient.IsNotAvailable(err) {
    ...
} else if err != nil {
    ....
} 

var nodes []corev1.Node
...
```

Reply:
Yeah, so I try to explain it.

The current code always checks if the tenantclient is available but this takes forever. So we immediately return and wait for the next reconcilation.

This means we cannot create the timestamp for the creating condition some lines below. So it will only set the timestamp until we have the tenantclient which is the time when we finished creating the cluster

So it will set creating and created with the same timestamp.

You can test any cluster with the latest operator version you will have the same timestamp on creating and created timestamp.

Reply:
Got it. I just think you could have fixed it by removing the single line that returned early so that the code just falls through.